wpq864: add nor config and read the mac addr from uboot env

stargieg opened the pull request:
- Add the nor flash config to dts. Fixes: openwrt#15476
- Read the mac addr from uboot env. Fixes: openwrt#15477 and openwrt#15478

Signed-off-by: Patrick Grimm <patrick@lunatiki.de>

neheb commented on target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts:
```diff
@@ -183,8 +183,11 @@
 				label = "0:APPSBLENV";
 				reg = <0x1180000 0x0080000>;
+				macaddr_ubootenv_ethaddr: ethaddr {};
+				macaddr_ubootenv_eth1addr: eth1addr {};
```
does this actually work?

IIRC @rmilecki didn't merge such a change yet.

stargieg replied:
Yes it works. Better than previous PR https://github.com/openwrt/openwrt/pull/15486

neheb replied:
Interesting. That means I can fix my other PRs to add custom names as well.

neheb replied:
I just looked at the code. No way this works: https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/drivers/nvmem/u-boot-env.c?h=next-20240805#n119

u_boot_env_read_post_process_ethaddr is what sets the mac address. It only gets called with ethaddr.

hauke replied:
`ethaddr` is supported by our driver, but `eth1addr` is not supported.

hauke commented on target/linux/ipq806x/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq8064-wpq864.dts:
```diff
@@ -404,8 +411,99 @@
 
-&flash {
-	compatible = "jedec,spi-nor";
+&gsbi5 {
```
Why do you extend the `gsbi5` node and not the `flash` node?

When you extend the `flash` node you have to delete the `partition@1` node like this:
```
/delete-node/ partition@1;
```
https://elixir.bootlin.com/linux/v6.6.52/source/arch/arm/boot/dts/qcom/qcom-ipq8064-v1.0.dtsi#L51

stargieg replied:
THX for review.
I check gsbi5 vs flash.
Should I add the BUG #15476 Description in the commit comment?

hauke replied:
yes please add a description why you change the code into the commit comment.

stargieg replied:
@hauke 
I have change the extend from gsbi5 node to the flash node and delete the partition@1 node. Is tested an working fine. :-)

stargieg committed:
ipq806x: add nvmem-cell mac-address for wpq864

Read the mac addr from uboot env.

You have to set these addr in the u-boot terminal or via
fw_setenv.
fw_setenv ethaddr 00:03:7f:ba:db:01
fw_setenv eth1addr 00:03:7f:ba:db:02

Ref: https://compexsystems.freshdesk.com/support/solutions/articles/36000179507-flash-lede-firmware-on-wpq864-wpq865

Fixes: openwrt#15477 and openwrt#15478
Signed-off-by: Patrick Grimm <patrick@lunatiki.de>

stargieg committed:
ipq806x: add nor config wpq864 Add the nor flash config to dts

Boot openwrt-ipq806x-generic-compex_wpq864-initramfs-fit-uImage.itb via tftpboot.
run sysupgrade -n openwrt-ipq806x-generic-compex_wpq864-squashfs-nand-sysupgrade.bin
next reboot halted with kernelpanic "Unable to mount root fs"

```
[    3.703628] ubi0: background thread "ubi_bgt0d" started,?[    3.717563] List of all partitions:
[    3.717603] 1f00             256 mtdblock0
[    3.717615]  (driver?)
[    3.724124] 1f01            1280 mtdblock1
[    3.724134]  (driver?)
[    3.730528] 1f02            1280 mtdblock2
[    3.730539]  (driver?)
[    3.737041] 1f03            2560 mtdblock3
[    3.737050]  (driver?)
[    3.743620] 1f04            1152 mtdblock4
[    3.743629]  (driver?)
[    3.750059] 1f05            1152 mtdblock5
[    3.750067]  (driver?)
[    3.756571] 1f06            2560 mtdblock6
[    3.756580]  (driver?)
[    3.763144] 1f07            2560 mtdblock7
[    3.763154]  (driver?)
[    3.769591] 1f08            5120 mtdblock8
[    3.769598]  (driver?)
[    3.776103] 1f09             512 mtdblock9
[    3.776112]  (driver?)
[    3.782669] 1f0a            1280 mtdblock10
[    3.782680]  (driver?)
[    3.789470] 1f0b           65536 mtdblock11
[    3.789477]  (driver?)
[    3.796068] 1f0c             384 mtdblock12
[    3.796077]  (driver?)
[    3.802726] 1f0d            1280 mtdblock13
[    3.802735]  (driver?)
[    3.809262] 1f0e            2560 mtdblock14
[    3.809270]  (driver?)
[    3.815859] 1f0f            1152 mtdblock15
[    3.815868]  (driver?)
[    3.822532] 1f10            1152 mtdblock16
[    3.822541]  (driver?)
[    3.829053] 1f11            2560 mtdblock17
[    3.829061]  (driver?)
[    3.835652] 1f12            2560 mtdblock18
[    3.835662]  (driver?)
[    3.842308] 1f13             384 mtdblock19
[    3.842317]  (driver?)
[    3.848844] 1f14            5120 mtdblock20
[    3.848853]  (driver?)
[    3.855443] 1f15           65536 mtdblock21
[    3.855455]  (driver?)
[    3.862113] 1f16           94208 mtdblock22
[    3.862122]  (driver?)
[    3.868635] 1f17           16384 mtdblock23
[    3.868643]  (driver?)
[    3.875235] 1f18           16384 mtdblock24
[    3.875243]  (driver?)
[    3.881888] fe00            8060 ubiblock0_1
[    3.881896]  (driver?)
[    3.888425] No filesystem could mount root, tried:
[    3.888431]  squashfs
[    3.890684]
[    3.897887] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(31,23)
```

Fixes: openwrt#15476

Signed-off-by: Patrick Grimm <patrick@lunatiki.de>
